R/draw_heatmap.R

```diff
@@ -0,0 +1,93 @@
+#' Heatmap for Gene Expression Counts
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' This produces a heatmap of the normalized gene expression counts and groups by various factors available in the assay dataset
+#'
+#' @param object (`AnyHermesData`)\cr input.
+#' @param assay_name (`string`)\cr selects assay from input.
+#' @param color_extremes (`numeric`)\cr percentiles of dataset intended to be displayed.
+#' @param col_data_annotation (`string` or `NULL`)\cr optional grouping variable, taken
+#'   from input sample variables.
+#' @param ...(`string` or `NULL` or `??`)\cr additional arguments to pass to the ComplexHeatmap function
+#'
+#' @return The `ComplexHeatmap::Heatmap` heatmap
+#'
+#' @export
+#'
+#' @examples
+#'
+#' result <- hermes_data %>%
+#'   normalize(methods = "voom") %>%
+#'   add_quality_flags()
+#'
+#' result2 <- hermes::filter(result, what = "genes")
```

Comment:
```suggestion
#' result <- hermes_data %>%
#'   normalize(methods = "voom") %>%
#'   add_quality_flags() %>%
#'   filter(what = "genes")
```
because you don't need two different results below, just work with this one

Reply:
also no need to prefix `filter` with the package name here, because we are in the package - the method for `HermesData` for the generic `filter()` function will be loaded